### ISSUE TITLE
Failing with wmi but working with windll

### ISSUE THREAD

**Issue by lcharles** [Crozzers/screen_brightness_control#1]
Hi first of all thank you for your job on this little library, I am looking at a few different brightness controllers (I could contribute too) and I thought I would give yours a try first to see how it works. 

Unfortunately I have to report that I could not make it work on windows 10 (2004), it raises an error here.
```
File "C:\...\miniconda3\envs\winrt\lib\site-packages\screen_brightness_control\__init__.py", line 157, in get_brightness 
screen_brightness_control.ScreenBrightnessError: Cannot retrieve screen brightness: monitor does not support it
```
 
I tried a few things in ipython to narrow down the issue and apparently line 139 raises the error: 
`brightness_method = wmi.WMI(namespace='wmi').WmiMonitorBrightnessMethods()`

The error does not seem to be a hardware limitation though here is the error that I get when I run the line above on ipython:
```
  ~\miniconda3\envs\winrt\lib\site-packages\win32com\client\util.py in __GetIndex(self, index)
     52                 self.index = index
---> 53                 result = self._oleobj_.Next(1)
     54                 if len(result):
com_error: (-2147217396, 'OLE error 0x8004100c', None, None)
```
The weird thing is that I can actually set the brightness if I follow this tutorial that uses ctypes instead of pywin32:
https://stackoverflow.com/questions/16588133/sending-ddc-ci-commands-to-monitor-on-windows-using-python

I was wondering, if maybe selected brands of monitors were supported by one method only.
Maybe adding this method as a fallback could be useful ?

I am very much willing to contribute, my ultimate goal is to be able to setup a little app in the notification area (for some basic settings) and setup two global keyboard shortcuts to change the brightness of the screen from the keyboard like on any laptop.

Having it to work on both windows and Linux at the same time would be great.
As for macOS support, I wouldn't worry about it, there are already good tools that do it.

**Comment by Crozzers** [comment]
Hello,
Thanks for raising this issue.

The `com_error` you reported is the same error I get when I use this on my desktop monitors, which have adjustable brightness but only through their on-screen-displays (not through Windows settings).
I wasn't aware of being able to send DDC CI commands before so I will look into this soon, however, after a quick glance it doesn't appear to be straight forward as to how the brightness can be altered (aside from turning the display off, as shown in the example). If the monitor brightness can be altered via DDC commands then I would be more than willing to add it as a fallback for Windows, and possibly Linux of they have something similar.

In terms of contribution, I'm open to contribution but bear in mind that the structure of the library is changing in the near future to allow for easier upgrades, such as this. [(see the dev branch)](https://github.com/Crozzers/screen_brightness_control/tree/dev)

As for OS support, I figured that MAC would have some nicely polished tool already so I've put that idea on hold for now.

**Comment by lcharles** [comment]
> Hello,
> Thanks for raising this issue.
> 
> The `com_error` you reported is the same error I get when I use this on my desktop monitors, which have adjustable brightness but only through their on-screen-displays (not through Windows settings).
> I wasn't aware of being able to send DDC CI commands before so I will look into this soon, however, after a quick glance it doesn't appear to be straight forward as to how the brightness can be altered (aside from turning the display off, as shown in the example). If the monitor brightness can be altered via DDC commands then I would be more than willing to add it as a fallback for Windows, and possibly Linux of they have something similar.

In the example you can switch the opcode 0xd6 in `set_vcp_feature(handle, 0xd6, 0x04)` to 0x10 to send the new brightness, the new value need to be between 0x00 and and 0x64.
But yeah it's hard to find a good doc on vcp codes, I mostly learned them by using ddcutil (see below) in Ubuntu a few years ago.
There is another ddc ci tool in python for windows but it looks like it is dead now. You can look at it for the finer details.
https://github.com/dot-osk/monitor_ctrl
In Linux there is ddcutil that works well https://www.ddcutil.com
Maybe switching to it might be more robust, I don't know how your lib works on Linux.
> 
> In terms of contribution, I'm open to contribution but bear in mind that the structure of the library is changing in the near future to allow for easier upgrades, such as this. [(see the dev branch)](https://github.com/Crozzers/screen_brightness_control/tree/dev)

I don't think this would a problem for me. I can just submit patches of code to solve specific problems and I'll let you decide how to merge it. And

**Comment by Crozzers** [comment]
I ran a quick test on my monitors and the brightness adjustment does work, however I noticed an odd issue where I could set the brightness and the change would show up in the on-screen-display but the backlight would not change untill I manually adjusted it. This was on my HDMI monitor, the DVI one worked fine. This may be an issue specific to the monitor then.

For Linux I have found a system function of `fnctl`, which would negate the need to add another system-install requirement so I shall look into that, but failing this ddcutil looks like a good option.
On Linux my library just cycles through a number of commands and hopes the user has the program installed because I couldn't find a general OS wide way to adjust brightness like `WMI` at the time.

I'm happy to merge any patches/contributions
